Tasks/ANT/anttask.ts

```diff
@@ -3,103 +3,18 @@
-}
+var isWindows = os.type().match(/^Win/);
 
-// update JAVA_HOME if user selected specific JDK version or set path manually
-var javaHomeSelection = tl.getInput('javaHomeSelection', true);
-var specifiedJavaHome = null;
-
-if (javaHomeSelection == 'JDKVersion') {
-    tl.debug('Using JDK version to find and set JAVA_HOME');
-    var jdkVersion = tl.getInput('jdkVersion');
-    var jdkArchitecture = tl.getInput('jdkArchitecture');
-
-    if (jdkVersion != 'default') {
-        // jdkVersion should be in the form of 1.7, 1.8, or 1.10
-        // jdkArchitecture is either x64 or x86
-        // envName for version 1.7 and x64 would be "JAVA_HOME_7_X64"
-        var envName = "JAVA_HOME_" + jdkVersion.slice(2) + "_" + jdkArchitecture.toUpperCase();
-        specifiedJavaHome = tl.getVariable(envName);
-        if (!specifiedJavaHome) {
-            tl.error('Failed to find specified JDK version. Please make sure environment variable ' + envName + ' exists and is set to the location of a corresponding JDK.');
-            tl.exit(1);
-        }
+function pathExistsAsFile(path: string) {
+    try {
+        return tl.stats(path).isFile();
+    }
+    catch (error) {
+        return false;
     }
 }
-else {
-    tl.debug('Using path from user input to set JAVA_HOME');
-    var jdkUserInputPath = tl.getPathInput('jdkUserInputPath', true, true);
-    specifiedJavaHome = jdkUserInputPath;
-}
-
-if (specifiedJavaHome) {
-    tl.debug('Set JAVA_HOME to ' + specifiedJavaHome);
-    process.env['JAVA_HOME'] = specifiedJavaHome;
-}
-
-var ccTool = tl.getInput('codeCoverageTool');
-var isCodeCoverageOpted = (typeof ccTool != "undefined" && ccTool && ccTool.toLowerCase() != 'none');
-
-var buildRootPath = path.dirname(antBuildFile);
-var instrumentedClassesDirectory = path.join(buildRootPath, "InstrumentedClasses");
-//delete any previous cobertura instrumented classes as they might interfere with ant execution.
-tl.rmRF(instrumentedClassesDirectory, true);
-
-if (isCodeCoverageOpted) {
-    var summaryFile: string = null;
-    var reportDirectory: string = null;
-    var ccReportTask: string = null;
-    var reportBuildFile: string = null;
-    enableCodeCoverage();
-}
-else {
-    tl.debug("Option to enable code coverage was not selected and is being skipped.");
-}
-
-var publishJUnitResults = tl.getInput('publishJUnitResults');
-var testResultsFiles = tl.getInput('testResultsFiles', true);
-
-antv.exec()
-    .then(function(code) {
-        return antb.exec();
-    })
-    .then(function(code) {
-        publishTestResults(publishJUnitResults, testResultsFiles);
-        publishCodeCoverage(isCodeCoverageOpted);
-        tl.exit(code);
-    })
-    .fail(function(err) {
-        publishTestResults(publishJUnitResults, testResultsFiles);
-        console.error(err.message);
-        tl.debug('taskRunner fail');
-        tl.exit(1);
-    })
 
 function publishTestResults(publishJUnitResults, testResultsFiles: string) {
```

Comment:
Can you move the helper functions into a separate file? If the implementation can be shared with maven or other tasks it might be worth moving into a common module